### ISSUE TITLE
Relatively High CPU Usage When Paused and not Minimised

### ISSUE THREAD

**Issue by jGleitz** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#2766]
First off, I have seen that there already are [a lot of issues concerning CPU usage](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues?utf8=%E2%9C%93&q=cpu%20usage). But I think that mine is different. 

**OS:** Ubuntu 17.04

**Issue Descriptions:**

Google Play Music Player has an (in my opinion) fairly low CPU usage when playing music:

![playing music](https://user-images.githubusercontent.com/4305652/30512910-d29465ba-9af9-11e7-982f-9d8e58685ba1.png)

Great work on that! :+1: 

However, the CPU usage is just as high (or higher!) when the music is paused:

![not playing music](https://user-images.githubusercontent.com/4305652/30512907-c938f4a4-9af9-11e7-833b-8fd930ee0288.png)

When taking this screenshot, the Google Play Music Desktop Player window was open on another workspace, but playback was paused. Minimising the window solves the issue. It does not affect the issue whether the window is open on current or another workspace.

If Google Play Music is opened in Chrome, the CPU usage is similar while playing music. But it goes down to 0 when music is paused, regardless of whether the window is visible or not.

To summarise, I’d expect the CPU usage of Google Play Music Desktop Player to drop to zero shortly after pausing playback.

(Please excuse my backup tool taking the first place in both screenshots).

**Steps to Reproduce:**

 * Open Google Play Music Desktop Player
 * Start some music.
 * Pause the music.
 * Wait a little for things to settle.
 * Look at the CPU usage
 * Minimise the window -> CPU usage goes to 0
 * Open the window again -> CPU usage is at 2-5% again.

**Comment by betorobson** [comment]
I can confirm that is true. In my new MacBook Pro 2017 I can se a high CPU usage e extremely high energy impact on Activity Monitor which turn the fans on. It is a very nice app however, It would have a good performance improvement.
Hope that you guys can fix it soon.
Going back to web chrome version :-(

**Comment by sliday** [comment]
@betorobson I'm with you on this one, on the new MacBook Pro 2017 too, had to uninstall GPMDP because of high CPU usage. Would be great to see this resolved.

**Comment by pamidur** [comment]
Same on Windows 10. Have always remember to minimize GPMDP so my laptop can make it till evening.

**Comment by joaohkfaria** [comment]
@yangwao I don't think that made any difference. Look:

Before:
<img width="797" alt="before" src="https://user-images.githubusercontent.com/1427347/32982937-74caca90-cc73-11e7-974f-46d3bd187abd.png">

After:
<img width="793" alt="after" src="https://user-images.githubusercontent.com/1427347/32982938-7c7e3a6a-cc73-11e7-871f-f56021347f5b.png">

It's only making a huge difference when I minimise the window.

**Comment by andbuitra** [comment]
Oh boy I haven't even noticed this before. 

It's the same for me on Linux (elementaryOS). Checking htop if I don't minimize GPMDP the CPU will stay at 50% ish. Even if I just change my current app it won't go down.

**Comment by andrewNestorenko** [comment]
Same here, Win 10, ~15% CPU Usage

**Comment by henkesn** [comment]
Same here, Kubuntu 17.10, KDE 5.12
~20% CPU unless minimized.

**Comment by markhamb** [comment]
OSX Sierra, MacBook Pro (Retina, 15-inch, Mid 2015).  I see 10-20% CPU when window open, but idle, and < 1% when minimized.  This is particularly for the "helper" process.  The parent does not seem so aggressive.

**Comment by multiflexi** [comment]
On Ubuntu 18.04 uses over 25% of CPU in 4th gen Intel i5 even when not playing anything and is very slow and unresponsive.

**Comment by keerah** [comment]
Guys, using this app is really too expensive. I compared playing music with Foobar with all sound processing and whistles takes about 0,3% on my system, but your app takes 1% even when the music is paused. This doesn't make me want to use it.

**Comment by multiflexi** [comment]
![screenshot from 2018-05-08 23-35-25](https://user-images.githubusercontent.com/11663844/39784967-afcb81f6-531a-11e8-9490-43f229fe7caa.png)
This is how much resources it uses, recently it completely froze GNOME, had to kill it. I really wish to use it by I simply cannot now.
